### PR TITLE
Implemented ResultStorageResult class

### PR DESCRIPTION
**Pull request by Bladrak**
As seen in #574 implemented a ResultStorageResult class to avoid doubling some calls

### CODE REVIEW

**Comment by Bladrak** [comment]
@masom does this look like what you were thinking?

File: thumbor/result_storages/__init__.py [hunk 1]
```diff
@@ -12,6 +12,26 @@
...
+from thumbor.loaders import LoaderResult
+from thumbor.engines import BaseEngine
+
+class ResultStorageResult(LoaderResult):
+    def last_modified(self):
+        '''
+        Retrieves last_updated metadata if available
+        :return:
+        '''
+        return self.metadata['LastModified'] if 'LastUpdated' in self.metadata else None
```
Comment (masom) [inline thumbor/result_storages/__init__.py]: Can just be `return self.metadata.get('LastModified', None)`

Comment (masom) [inline thumbor/result_storages/__init__.py]: Also `LastUpdated != LastModified`

File: thumbor/result_storages/__init__.py [hunk 2]
```diff
@@ -12,6 +12,26 @@
...
+
+    def mime(self):
+        '''
+        Retrieves mime metadata if available
+        :return:
+        '''
+        return self.metadata['ContentType'] if 'ContentType' in self.metadata else BaseEngine.get_mimetype(self.buffer)
```
Comment (masom) [inline thumbor/result_storages/__init__.py]: Can just be `return self.metadata.get('ContentType', BaseEngine.get_mimetype(self.buffer))`

File: thumbor/result_storages/file_storage.py [hunk 3]
```diff
@@ -55,11 +57,12 @@ def get(self, callback):
...
         else:
-            buffer = None
+            result = ResultStorageResult()
 
             with open(file_abspath, 'r') as f:
-                buffer = f.read()
-            callback(buffer)
+                result.buffer = f.read()
+
+            callback(result)
```
Comment (masom) [inline thumbor/result_storages/file_storage.py]: Before returning the result the file_storage should set some metadata like what is being done in `last_updated`.

**Comment by Bladrak** [comment]
fixed @masom

**Comment by heynemann** [comment]
This looks Backwards compatible. I'm just asking to be safe. Anyone tested this to make sure thumbor keeps working as it does today?

**Comment by Bladrak** [comment]
I tested it with the aws result storage before adapting it to this edit, and it worked fine :)

**Comment by Bladrak** [comment]
@masom @heynemann added some tests as well

**Comment by masom** [comment]
@Bladrak i'll test this PR on PopKey's internal servers and report back if anything broke.

**Comment by masom** [comment]
@Bladrak 
Getting this:
```
  File "/Users/martin/venv/thumbor/lib/python2.7/site-packages/tornado/gen.py", line 230, in wrapper
    yielded = next(result)
  File "/Users/martin/dev/os/thumbor/thumbor/handlers/__init__.py", line 265, in _process_result_from_storage
    result_last_modified = result.last_modified()
TypeError: 'datetime.datetime' object is not callable
```

**Comment by masom** [comment]
It's triggering that because `last_modified` is a property, as defined in https://github.com/thumbor/thumbor/pull/575/files#diff-69a17e56e54662941df7e762f02f2b2cR19

**Comment by Bladrak** [comment]
@masom my bad, updated to fix this :)

**Comment by masom** [comment]
:+1:

**Comment by masom** [comment]
I'll be doing a last test on a storage ( redis ) that hasn't migrated to ensure this will be backward compatible

**Comment by Bladrak** [comment]
Ok, thanks @masom :)

**Comment by masom** [comment]
Works! Thanks @Bladrak

**Comment by Bladrak** [comment]
Thanks @masom :)
@guilhermef @heynemann is it possible to issue a new version on pypi? So I can release the aws connector update and specify its dependency towards this update in thumbor in the setup.py file :)

**Comment by guilhermef** [comment]
sure @Bladrak, I'll do it now.

**Comment by Bladrak** [comment]
thanks @guilhermef ! :+1:

**Comment by guilhermef** [comment]
Done @Bladrak https://github.com/thumbor/thumbor/releases/tag/5.2.0

**Comment by Bladrak** [comment]
Great :) Thanks again!

**Pull request merged**